subsys/net/openthread/Kconfig.defconfig

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# The purpose of this file is to define new default values of settings related
+# to building Thread from sources.
+# This file only changes defaults and thus all symbols here must be prompltless
+# and safeguarded so that they only are applied when building Thread.
+
+if NET_L2_OPENTHREAD
+
+choice LIBC_IMPLEMENTATION
+	default NEWLIB_LIBC
+endchoice
+
+config NUM_METAIRQ_PRIORITIES
+	int
+	default 1
+
+config SYSTEM_WORKQUEUE_STACK_SIZE
+	int
+	default 1120
+
+config HW_STACK_PROTECTION
+	bool
+	default y
+
+config NET_CONNECTION_MANAGER
+	bool
+	default y
+
+config MAIN_STACK_SIZE
+	int
+	default 2048
```

Comment:
Why and when is a stack size of `2048` in the cases where it's not already `2048` per default ?

stack size is already 2048 when using cc3xx entropy:
https://github.com/nrfconnect/sdk-nrf/blob/612928cec9588b41357139771ae2046ad7183e8c/Kconfig.nrf#L36

Which is also default here:
https://github.com/nrfconnect/sdk-nrf/blob/612928cec9588b41357139771ae2046ad7183e8c/drivers/entropy/Kconfig#L7-L12